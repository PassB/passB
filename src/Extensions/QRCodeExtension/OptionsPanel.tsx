import {FormControl, FormControlLabel, FormLabel, Grid, Input, List, ListItem, Radio, RadioGroup} from 'material-ui';
import {withStyles, ClassProps} from 'material-ui/styles';
import * as React from 'react';
import {QRCode} from 'react-qr-svg';
import {OptionPanelProps} from 'Options/OptionsReceiver';
import {Level, Options} from './QRCodeExtension';

const styles = {
  qrcode: {
    height: '150px',
  },
};

export const OptionsPanel = withStyles<OptionPanelProps<Options>>(styles)(
  ({options, updateOptions, classes}: OptionPanelProps<Options> & ClassProps<typeof styles>): JSX.Element => (
    <Grid container={true} direction="row" justify="space-between" align="center" spacing={0}>
      <Grid item={true}>
        <List>
          <ListItem>
            <FormControl>
              <FormLabel>Error correction level:</FormLabel>
              <RadioGroup
                value={options.level}
                row={true}
                onChange={(event: React.InputHTMLAttributes<HTMLInputElement>, value: Level) => updateOptions({
                  ...options,
                  level: value,
                })}
              >

                <FormControlLabel value="L" control={<Radio/>} label="L"/>
                <FormControlLabel value="M" control={<Radio/>} label="M"/>
                <FormControlLabel value="Q" control={<Radio/>} label="Q"/>
                <FormControlLabel value="H" control={<Radio/>} label="H"/>

              </RadioGroup>
            </FormControl>
          </ListItem>
          <ListItem>
            <FormControl>
              <FormLabel>Background color:</FormLabel>
              <Input
                inputProps={{
                  onChange: (event: React.ChangeEvent<HTMLInputElement>) => updateOptions({
                    ...options,
                    bgColor: event.target.value,
                  }),
                }}
                type="color"
                value={options.bgColor}
              />
            </FormControl>
          </ListItem>
          <ListItem>
            <FormControl>
              <FormLabel>Foreground color:</FormLabel>
              <Input
                inputProps={{
                  onChange: (event: React.ChangeEvent<HTMLInputElement>) => updateOptions({
                    ...options,
                    fgColor: event.target.value,
                  }),
                }}
                type="color"
                value={options.fgColor}
              />
            </FormControl>
          </ListItem>
        </List>
      </Grid>
      <QRCode
        className={classes.qrcode}
        {...options}
        value="example. happy testing ðŸ˜€"
      />
    </Grid>
  ));
