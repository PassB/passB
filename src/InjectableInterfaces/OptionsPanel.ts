import {ComponentType} from 'react';
import {MapTypeAllowedData, TypedMap} from 'State/Types/TypedMap';

export interface OptionPanelProps<OptionType extends MapTypeAllowedData<OptionType>> {
  options: TypedMap<OptionType>;
  updateOptions: (newOptions: TypedMap<OptionType>) => void;
}

export type OptionsPanelType<OptionType extends MapTypeAllowedData<OptionType>> = ComponentType<OptionPanelProps<OptionType>>;
