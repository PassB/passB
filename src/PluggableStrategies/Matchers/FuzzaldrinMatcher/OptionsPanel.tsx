import {TextField} from 'material-ui';
import * as React from 'react';
import {OptionPanelProps} from 'Options/OptionsReceiver';
import {Options} from './FuzzaldrinMatcher';

export function OptionsPanel({options, updateOptions}: OptionPanelProps<Options>): JSX.Element {
  return (
    <div>
      <TextField
        label="Minimum score:"
        helperText=""
        value={options.minScore}
        onChange={(e: any) => updateOptions({ // tslint:disable-line:no-any
          // required until material-ui get their typings right
          // this will be finally fixed with beta v1.0.0-beta.17
          ...options,
          minScore: Math.max(0, (e as React.ChangeEvent<HTMLInputElement>).target.valueAsNumber),
        })}
        type="number"
      />
      <br/>
      <TextField
        label="Maximum visible results:"
        helperText="to show all results, enter '0'"
        value={options.maxResults}
        onChange={(e: any) => updateOptions({ // tslint:disable-line:no-any
          // required until material-ui get their typings right
          ...options,
          maxResults: Math.max(0, (e as React.ChangeEvent<HTMLInputElement>).target.valueAsNumber),
        })}
        type="number"
      />
    </div>
  );
}
