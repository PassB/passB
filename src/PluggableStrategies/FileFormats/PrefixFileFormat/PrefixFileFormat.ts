import {Service} from 'typedi';
import {OptionsPanelType} from 'Options/OptionsReceiver';
import {createTypedMap} from 'State/Types/TypedMap';
import {FileFormat, FileFormatTag} from '../';
import {OptionsPanel} from './OptionsPanel';

export interface Options {
  passwordFirstLine: boolean;
  passwordPrefix: string;
  usernamePrefix: string;
  trimWhitespace: boolean;
}

@Service({tags: [FileFormatTag]})
export class PrefixFileFormat extends FileFormat<Options> {
  public readonly OptionsPanel: OptionsPanelType<Options> = OptionsPanel;

  public constructor() {
    super(
      'PrefixFileFormat',
      createTypedMap({
        passwordFirstLine: true,
        passwordPrefix: '',
        usernamePrefix: 'login:',
        trimWhitespace: true,
      }),
    );
  }

  public getPassword(lines: string[], entryName: string): string | undefined {
    if (this.options.get('passwordFirstLine')) {
      return lines[0];
    }
    return this.getEntryByPrefix(lines, this.options.get('passwordPrefix'));
  }

  public getUsername(lines: string[], entryName: string): string | undefined {
    return this.getEntryByPrefix(lines, this.options.get('usernamePrefix'));
  }

  private getEntryByPrefix(lines: string[], prefix: string): string | undefined {
    for (const line of lines) {
      if (line.startsWith(prefix)) {
        let entry = line.substr(prefix.length);
        if (this.options.get('trimWhitespace')) {
          entry = entry.trim();
        }
        return entry;
      }
    }
    return;
  }
}
