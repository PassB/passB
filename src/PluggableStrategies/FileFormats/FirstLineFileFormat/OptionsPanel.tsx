import {FormControl, FormControlLabel, FormLabel} from 'material-ui';
import {default as Radio, RadioGroup} from 'material-ui/Radio';
import * as React from 'react';
import {OptionPanelProps} from 'InjectableInterfaces/OptionsPanel';
import {Options, UsernameStyle} from './FirstLineFileFormat';

export function OptionsPanel({options, updateOptions}: OptionPanelProps<Options>): JSX.Element {
  return (
    <div>
      <FormControl>
        <FormLabel>Get username from:</FormLabel>
        <RadioGroup
          name="FirstLineFileFormat_usernameStyle"
          value={options.get('usernameStyle')}
          onChange={(event: React.InputHTMLAttributes<HTMLInputElement>, value: UsernameStyle) =>
            updateOptions(options.set('usernameStyle', value))
          }
        >
          <FormControlLabel value="None" control={<Radio/>} label="None"/>
          <FormControlLabel value="SecondLine" control={<Radio/>} label="SecondLine"/>
          <FormControlLabel value="LastPathPart" control={<Radio/>} label="LastPathPart"/>
        </RadioGroup>
      </FormControl>
    </div>
  );
}
