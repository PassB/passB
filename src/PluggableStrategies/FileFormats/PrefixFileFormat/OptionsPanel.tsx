import {Checkbox, FormControl, FormControlLabel, TextField} from 'material-ui';
import * as React from 'react';
import {OptionPanelProps} from 'InjectableInterfaces/OptionsPanel';
import {Options} from './PrefixFileFormat';

export function OptionsPanel({options, updateOptions}: OptionPanelProps<Options>): JSX.Element {
  return (
    <FormControl fullWidth={true}>
      <FormControlLabel
        id="passwordFirstLine"
        control={
          <Checkbox
            checked={options.get('passwordFirstLine')}
            onChange={
              (event: React.ChangeEvent<HTMLInputElement>, isChecked: boolean) => updateOptions(options.set('passwordFirstLine', isChecked))
            }
          />
        }
        label="Password is in first line (without prefix)"
      />
      {!options.get('passwordFirstLine') &&
      <TextField
        id="passwordPrefix"
        label="password prefix"
        value={options.get('passwordPrefix')}
        onChange={
          (event: React.ChangeEvent<HTMLInputElement>) => updateOptions(options.set('passwordPrefix', event.target.value))
        }
      />
      }
      <TextField
        id="usernamePrefix"
        label="username prefix"
        value={options.get('usernamePrefix')}
        onChange={
          (event: React.ChangeEvent<HTMLInputElement>) => updateOptions(options.set('usernamePrefix', event.target.value))
        }
      />
      <FormControlLabel
        id="trimWhitespace"
        control={
          <Checkbox
            checked={options.get('trimWhitespace')}
            onChange={
              (event: React.ChangeEvent<HTMLInputElement>, isChecked: boolean) =>  updateOptions(options.set('trimWhitespace', isChecked))
            }
          />
        }
        label="Trim whitespace"
      />
    </FormControl>
  );
}
