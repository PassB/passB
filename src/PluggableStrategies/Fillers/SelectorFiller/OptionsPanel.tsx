import {Checkbox, FormControl, FormControlLabel, TextField, Typography, WithStyles} from 'material-ui';
import withStyles from 'material-ui/styles/withStyles';
import * as React from 'react';
import {OptionPanelProps} from 'InjectableInterfaces/OptionsPanel';
import {Options} from './SelectorFiller';

const styles = {
  topPadding: {
    paddingTop: '24px',
  },
};

interface State {
  invalidProps: {
    defaultUsernameSelector?: string | false;
    defaultPasswordSelector?: string | false;
  };
}

class OptionsPanelComponent extends React.Component<OptionPanelProps<Options> & WithStyles<keyof typeof styles>, State> {
  public state: State = {
    invalidProps: {},
  };

  public render(): JSX.Element {
    const {
      options,
      updateOptions,
      classes,
    } = this.props;

    const {
      invalidProps: {
        defaultUsernameSelector: invalidDefaultUsernameSelector,
        defaultPasswordSelector: invalidDefaultPasswordSelector,
      },
    } = this.state;

    const {
      defaultUsernameSelector,
      defaultPasswordSelector,
      fillOnlyFirstMatchingField,
      usernameSelectorPrefix,
      passwordSelectorPrefix,
    } = options.toJS();

    return (
      <FormControl fullWidth={true}>
        <TextField
          id="defaultUsernameSelector"
          error={!!invalidDefaultUsernameSelector}
          label={browser.i18n.getMessage('options_selectorfiller_defaultUsernameSelector')}
          helperText={browser.i18n.getMessage('options_selectorfiller_defaultUsernameSelector_helperText')}
          value={invalidDefaultUsernameSelector || defaultUsernameSelector}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) => this.setSelectorOptions('defaultUsernameSelector', event.target.value)}
          margin="normal"
        />
        <TextField
          id="defaultPasswordSelector"
          error={!!invalidDefaultPasswordSelector}
          label={browser.i18n.getMessage('options_selectorfiller_defaultPasswordSelector')}
          helperText={browser.i18n.getMessage('options_selectorfiller_defaultPasswordSelector_helperText')}
          value={invalidDefaultPasswordSelector || defaultPasswordSelector}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) => this.setSelectorOptions('defaultPasswordSelector', event.target.value)}
          margin="normal"
        />
        <FormControlLabel
          id="fillOnlyFirstMatchingField"
          label={browser.i18n.getMessage('options_selectorfiller_fillOnlyFirstMatchingField')}
          control={
            <Checkbox
              checked={fillOnlyFirstMatchingField}
              onChange={
                (event: React.ChangeEvent<HTMLInputElement>, isChecked: boolean) =>
                  updateOptions(options.set('fillOnlyFirstMatchingField', isChecked))
              }
            />
          }
        />
        <Typography type="title" className={classes.topPadding} gutterBottom={true}>
          {browser.i18n.getMessage('options_selectorfiller_prefix_heading')}
        </Typography>
        <Typography type="subheading">
          {browser.i18n.getMessage('options_selectorfiller_prefix_explanation')}
        </Typography>
        <TextField
          id="usernameSelectorPrefix"
          label={browser.i18n.getMessage('options_selectorfiller_usernameSelectorPrefix')}
          value={usernameSelectorPrefix}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
            updateOptions(options.set('usernameSelectorPrefix', event.target.value))}
          margin="normal"
        />
        <TextField
          id="passwordSelectorPrefix"
          label={browser.i18n.getMessage('options_selectorfiller_passwordSelectorPrefix')}
          value={passwordSelectorPrefix}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
            updateOptions(options.set('passwordSelectorPrefix', event.target.value))}
          margin="normal"
        />
      </FormControl>
    );
  }

  private setSelectorOptions(selector: keyof Options, value: string): void {
    try {
      if (value) {
        // try to use the selector to detect invalid syntax
        document.querySelector(value);
      }
      this.props.updateOptions(this.props.options.set(selector, value));
      this.setState({invalidProps: {...this.state.invalidProps, [selector]: false}});
    } catch (e) {
      this.setState({invalidProps: {...this.state.invalidProps, [selector]: value}});
    }

  }
}

export const OptionsPanel = withStyles(styles)(OptionsPanelComponent);
