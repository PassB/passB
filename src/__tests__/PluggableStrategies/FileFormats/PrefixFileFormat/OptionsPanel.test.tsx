import {mount, shallow, ShallowWrapper} from 'enzyme';
import * as React from 'react';
import {OptionsPanel} from 'PluggableStrategies/FileFormats/PrefixFileFormat/OptionsPanel';
import {Options} from 'PluggableStrategies/FileFormats/PrefixFileFormat/PrefixFileFormat';
import {createTypedMap, TypedMap} from 'State/Types/TypedMap';

const getOptions = (partial: Partial<Options> = {}) => createTypedMap({
  passwordFirstLine: true,
  passwordPrefix: '',
  usernamePrefix: 'login:',
  trimWhitespace: true,
  ...partial,
});

describe('PrefixFileFormat OptionsPanel', () => {
  describe('rendering', () => {
    function testRendering(testOptions: TypedMap<Options>): void {
      let panel: ShallowWrapper<{}>;
      it('renders', () => {
        panel = shallow(
          <OptionsPanel options={testOptions} updateOptions={() => 0}/>,
        );
        expect(panel).toMatchSnapshot();
      });

      it('passwordFirstLine is checked correctly', () => {
        expect(
          shallow(panel.find('#passwordFirstLine').prop('control'))
            .prop('checked'),
        ).toBe(testOptions.get('passwordFirstLine'));
      });

      if (testOptions.get('passwordFirstLine')) {
        it('passwordPrefix is absent', () => {
          expect(panel.find('#passwordPrefix').length).toBe(0);
        });
      } else {
        it('passwordPrefix has correct value', () => {
          expect(panel.find('#passwordPrefix').prop('value')).toBe(testOptions.get('passwordPrefix'));
        });
      }

      it('usernamePrefix has correct value', () => {
        expect(panel.find({id: 'usernamePrefix'}).prop('value')).toBe(testOptions.get('usernamePrefix'));
      });

      it('trimWhitespace is checked correctly', () => {
        expect(
          shallow(panel.find('#trimWhitespace').prop('control'))
            .prop('checked'),
        ).toBe(testOptions.get('trimWhitespace'));
      });
    }

    describe('default values', () => testRendering(getOptions()));
    describe('without passwordPrefix', () => testRendering(getOptions({passwordFirstLine: false})));
    describe('completely different options', () => testRendering(getOptions({
      passwordFirstLine: false,
      passwordPrefix: 'pw:',
      usernamePrefix: 'lg:',
      trimWhitespace: false,
    })));
  });

  describe('should update options on interaction', () => {
    const optionsCallback = jest.fn();

    const testOptions: TypedMap<Options> = getOptions({
      passwordFirstLine: false,
      passwordPrefix: 'password:',
      usernamePrefix: 'login:',
      trimWhitespace: true,
    });

    const panel = mount(
      <OptionsPanel options={testOptions} updateOptions={optionsCallback}/>,
    );

    beforeEach(() => {
      optionsCallback.mockReset();
    });

    it('updates passwordFirstLine', () => {
      panel.find('#passwordFirstLine').find('input').hostNodes()
        .simulate('change', {target: {checked: !testOptions.get('passwordFirstLine')}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('passwordFirstLine', !testOptions.get('passwordFirstLine')));
    });

    it('updates passwordPrefix', () => {
      panel.find('#passwordPrefix').hostNodes().simulate('change', {target: {value: 'test'}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('passwordPrefix', 'test'));
    });

    it('updates usernamePrefix', () => {
      panel.find('#usernamePrefix').hostNodes().simulate('change', {target: {value: 'test'}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('usernamePrefix', 'test'));
    });

    it('updates trimWhitespace', () => {
      panel.find('#trimWhitespace').find('input').hostNodes()
        .simulate('change', {target: {checked: !testOptions.get('trimWhitespace')}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('trimWhitespace', !testOptions.get('trimWhitespace')));
    });
  });

});
