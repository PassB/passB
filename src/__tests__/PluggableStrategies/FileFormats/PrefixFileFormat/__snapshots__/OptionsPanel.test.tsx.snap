// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrefixFileFormat OptionsPanel rendering completely different options renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanel
    options={
      Immutable.Map {
        "passwordFirstLine": false,
        "passwordPrefix": "pw:",
        "usernamePrefix": "lg:",
        "trimWhitespace": false,
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="passwordFirstLine"
          label="Password is in first line (without prefix)"
        />,
        <TextField
          id="passwordPrefix"
          label="password prefix"
          onChange={[Function]}
          required={false}
          select={false}
          value="pw:"
        />,
        <TextField
          id="usernamePrefix"
          label="username prefix"
          onChange={[Function]}
          required={false}
          select={false}
          value="lg:"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="trimWhitespace"
          label="Trim whitespace"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "passwordFirstLine",
          "label": "Password is in first line (without prefix)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "passwordPrefix",
          "label": "password prefix",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "pw:",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "usernamePrefix",
          "label": "username prefix",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "lg:",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "trimWhitespace",
          "label": "Trim whitespace",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="passwordFirstLine"
            label="Password is in first line (without prefix)"
          />,
          <TextField
            id="passwordPrefix"
            label="password prefix"
            onChange={[Function]}
            required={false}
            select={false}
            value="pw:"
          />,
          <TextField
            id="usernamePrefix"
            label="username prefix"
            onChange={[Function]}
            required={false}
            select={false}
            value="lg:"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="trimWhitespace"
            label="Trim whitespace"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "passwordFirstLine",
            "label": "Password is in first line (without prefix)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "passwordPrefix",
            "label": "password prefix",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "pw:",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "usernamePrefix",
            "label": "username prefix",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "lg:",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "trimWhitespace",
            "label": "Trim whitespace",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PrefixFileFormat OptionsPanel rendering default values renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanel
    options={
      Immutable.Map {
        "passwordFirstLine": true,
        "passwordPrefix": "",
        "usernamePrefix": "login:",
        "trimWhitespace": true,
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="passwordFirstLine"
          label="Password is in first line (without prefix)"
        />,
        false,
        <TextField
          id="usernamePrefix"
          label="username prefix"
          onChange={[Function]}
          required={false}
          select={false}
          value="login:"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="trimWhitespace"
          label="Trim whitespace"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "passwordFirstLine",
          "label": "Password is in first line (without prefix)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "usernamePrefix",
          "label": "username prefix",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "login:",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "trimWhitespace",
          "label": "Trim whitespace",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="passwordFirstLine"
            label="Password is in first line (without prefix)"
          />,
          false,
          <TextField
            id="usernamePrefix"
            label="username prefix"
            onChange={[Function]}
            required={false}
            select={false}
            value="login:"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="trimWhitespace"
            label="Trim whitespace"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "passwordFirstLine",
            "label": "Password is in first line (without prefix)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "usernamePrefix",
            "label": "username prefix",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "login:",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "trimWhitespace",
            "label": "Trim whitespace",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PrefixFileFormat OptionsPanel rendering without passwordPrefix renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanel
    options={
      Immutable.Map {
        "passwordFirstLine": false,
        "passwordPrefix": "",
        "usernamePrefix": "login:",
        "trimWhitespace": true,
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="passwordFirstLine"
          label="Password is in first line (without prefix)"
        />,
        <TextField
          id="passwordPrefix"
          label="password prefix"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
        />,
        <TextField
          id="usernamePrefix"
          label="username prefix"
          onChange={[Function]}
          required={false}
          select={false}
          value="login:"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="trimWhitespace"
          label="Trim whitespace"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "passwordFirstLine",
          "label": "Password is in first line (without prefix)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "passwordPrefix",
          "label": "password prefix",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "usernamePrefix",
          "label": "username prefix",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "login:",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "trimWhitespace",
          "label": "Trim whitespace",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="passwordFirstLine"
            label="Password is in first line (without prefix)"
          />,
          <TextField
            id="passwordPrefix"
            label="password prefix"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
          />,
          <TextField
            id="usernamePrefix"
            label="username prefix"
            onChange={[Function]}
            required={false}
            select={false}
            value="login:"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="trimWhitespace"
            label="Trim whitespace"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "passwordFirstLine",
            "label": "Password is in first line (without prefix)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "passwordPrefix",
            "label": "password prefix",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "usernamePrefix",
            "label": "username prefix",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "login:",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "trimWhitespace",
            "label": "Trim whitespace",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
